src/components/Tables/FilterAccordion.tsx

```diff
@@ -0,0 +1,106 @@
+"use client";
+
+import React from "react";
+import { Accordion, Checkbox, AccordionSummary, AccordionDetails } from "@mui/material";
+import styled from "styled-components";
+import { faChevronDown } from "@fortawesome/free-solid-svg-icons";
+import Icon from "@/components/Icons/Icon";
+
+interface FilterAccordionProps {
+    toggleableHeaders?: string[];
+    shownHeaderKeys: string[];
+    setShownHeaderKeys: (headers: string[]) => void;
+    headers: [string, string][];
+}
+
+const Styling = styled.div`
+    flex-grow: 1;
+    height: 3rem;
+    overflow: visible;
+    z-index: 2;
+    margin-bottom: 0.5rem;
+    margin-top: 0.5rem;
+    & .MuiPaper-root {
+        background-color: ${(props) => props.theme.surfaceBackgroundColor};
+        border: 1px solid ${(props) => props.theme.secondaryBackgroundColor};
+        box-shadow: 0 0 1px ${(props) => props.theme.secondaryBackgroundColor};
+        // TODO: VFB16 change to a theme specific grey
```

Comment:
TODO Format:
`// TODO VFB-XXX Short description which is capitalised`
(sorry, I'm a little particular about this sort of thing!)